Bottom: ae89cbceb6c9e38ea49b04db0b83f9a351b75eb8
Top:    66522de14c2912c83e6499db689a35f398c43a9c
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-08 09:34:04 +0800

Refresh of autofs-5.0.7-fix-potential-null-dereference-in-lookup_mount.patch

---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index eb6e9dc..4b4ee89 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1130,7 +1130,9 @@ do_cache_lookup:
 	ret = ctxt->parse->parse_mount(ap, key, key_len,
 				       mapent, ctxt->parse->context);
 	if (ret) {
+		cache_writelock(mc);
 		cache_update_negative(mc, source, key, ap->negative_timeout);
+		cache_unlock(mc);
 		return NSS_STATUS_TRYAGAIN;
 	}
